app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -518,14 +518,15 @@ def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
 
-    existing_providers = Endpoint.where(:hostname => hostname.downcase)
-                                 .where.not(:resource_id => id).includes(:resource)
-                                 .select do |endpoint|
-                                   unless endpoint.resource.nil?
-                                     endpoint.resource.uid_ems == keystone_v3_domain_id &&
-                                       endpoint.resource.provider_region == provider_region
```

Comment:
Since all the rest of these checks are off of endpoint.resource I thought a `self.class.join(:endpoints)` would be cleaner